modules/battery_manager/php/testprovisioner.class.inc

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * This file implements a data provisioner to get all possible tests
+ * for the battery_manager menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\battery_manager;
+
+/**
+ * This class implements a data provisioner to get all possible tests
+ * for the battery_manager menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class TestProvisioner extends \LORIS\Data\Provisioners\DBRowProvisioner
+{
+    /**
+     * Create a Test Instance, which get rows fro mthe test_battery table.
+     */
+    function __construct()
+    {
+        parent::__construct(
+            "SELECT b.id,
+               t.Test_name as testName,
+               b.AgeMinDays as ageMinDays,
+               b.AgeMaxDays as ageMaxDays,
+               b.Stage as stage,
+               s.SubprojectID as subproject,
+               b.Visit_label as visitLabel,
+               p.CenterID as centerId,
+               b.firstVisit,
+               b.instr_order as instrumentOrder,
+               b.Active as active
+        FROM test_battery b
+          LEFT JOIN test_names t USING (Test_name)
```

Comment:
@HenriRabalais I tested this on my VM. Everything works perfect except I got some blank entries in Instrument column. I did a quick research on this and found that my test battery table have some instruments which are not in the test name table. (Not valid test battery entries actually). **This might not happen on a clean test battery table.** 
But I think as every valid instrument should have an entry in test_names, can we change the `LEFT JOIN test_names t USING (Test_name)` to `JOIN test_names t USING (Test_name)`

![NULL_entry](https://user-images.githubusercontent.com/23702452/54624833-87516200-4a44-11e9-8cde-a92cbc99d3ad.png)

@ridz Suggestions?

Reply:
@sruthymathew123 I think changing the join would definitely solve the problem but not changing it would actually help detect the problem... so I'm in between...

@driusan ??

Reply:
I think that makes sense

Reply:
@driusan which part, keeping the LEFT or removing it ?

Reply:
Removing it to exclude invalid instruments.